### PR TITLE
Implement SquashDict hint

### LINKED ISSUE #295: SquashDict
Processes a series of dictionary accesses, sorts keys, and determines if any keys are beyond a certain bound, preparing for dictionary compression.

Reference: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/squash_dict.cairo#L39

Depends on #245 for the usage of `globals()`

### PR DESCRIPTION
**Pull request by har777**
Fixes #295 

Will change from draft PR to open PR once other hints needed by the integration tests are merged.

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 1]
```diff
@@ -367,6 +368,157 @@ func createDictUpdateHinter(resolver hintReferenceResolver) (hinter.Hinter, erro
...
+			if err != nil {
+				return err
+			}
+
+			//> ids.first_key = key = keys.pop()
+			firstKeyAddr, err := firstKey.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			firstKeyValue := keys[len(keys)-1]
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:476]: use `.Pop()` from arrays utils?

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 2]
```diff
@@ -367,6 +368,157 @@ func createDictUpdateHinter(resolver hintReferenceResolver) (hinter.Hinter, erro
 	return newDictUpdateHint(dictPtr, key, newValue, prevValue), nil
 }
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:370]: add annotation on what it does and the args

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 3]
```diff
@@ -367,6 +368,157 @@ func createDictUpdateHinter(resolver hintReferenceResolver) (hinter.Hinter, erro
...
 
+func newSquashDictHint(dictAccesses, ptrDiff, nAccesses, bigKeys, firstKey hinter.ResOperander) hinter.Hinter {
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:371]: add annotation on what it does and the args

File: pkg/hintrunner/zero/zerohint_dictionaries_test.go [hunk 4]
```diff
@@ -515,5 +516,209 @@ func TestZeroHintDictionaries(t *testing.T) {
...
+					{Name: "dict_accesses.1.key", Kind: apRelative, Value: feltUint64(8)},
+					{Name: "dict_accesses.1.prev_value", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "dict_accesses.1.new_value", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "dict_accesses.2.key", Kind: apRelative, Value: feltUint64(1)},
+					{Name: "dict_accesses.2.prev_value", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "dict_accesses.2.new_value", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "dict_accesses.3.key", Kind: apRelative, Value: feltUint64(21)},
+					{Name: "dict_accesses.3.prev_value", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "dict_accesses.3.new_value", Kind: apRelative, Value: feltUint64(0)},
+					// largest key within range_check_builtin.bound
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_dictionaries_test.go:582]: what does it mean?

Comment (har777) [reply pkg/hintrunner/zero/zerohint_dictionaries_test.go:644]: ah its just noting that the largest key value is within a bound. the value of `ids.big_keys` depend on the outcome of this condition.

**Review by TAdev0 — APPROVED** [review]
lgtm

**Pull request merged**